src/lrm.rs

```diff
@@ -0,0 +1,382 @@
+            scale_position,
+            curve_position,
+        }
+    }
+
+    /// Build an unnamed anchor
+    pub fn new_unnamed(scale_position: ScalePosition, curve_position: CurvePosition) -> Self {
+        Self {
+            id: None,
+            scale_position,
+            curve_position,
+        }
+    }
+
+    fn as_named(&self) -> Option<NamedAnchor> {
+        self.id.as_ref().map(|id| NamedAnchor {
+            id: id.to_owned(),
+            scale_position: self.scale_position,
+            curve_position: self.curve_position,
+        })
+    }
+}
+
+// Private struct to be used when we only deal with anchor that have names
+struct NamedAnchor {
+    id: String,
+    scale_position: ScalePosition,
+    curve_position: CurvePosition,
+}
+
+/// A helper to build a scale by adding consecutives anchors with relative distances
+/// When having all the anchors and their distances in both scale and real position,
+/// it is simpler to directly build the LrmScale from an Vec<Anchor>
+/// Using the builder will however ensure that the scale is valid
+pub struct ScaleBuilder {
+    anchors: Vec<Anchor>,
+}
+
+impl ScaleBuilder {
+    /// Create a new scale with an initial anchor
+    pub fn new(anchor: Anchor) -> Self {
+        Self {
+            anchors: vec![anchor],
+        }
+    }
+
+    /// Builds a named anchor and adds it to the scale builder
+    /// Distances are relative to previous anchor
+    pub fn add_named(self, id: &str, scale_dist: ScalePosition, curve_dist: CurvePosition) -> Self {
+        self.add(Some(id.to_owned()), scale_dist, curve_dist)
+    }
+
+    /// Builds an unnamed anchor and adds it to the scale builder
+    /// Distances are relative to previous anchor
+    pub fn add_unnamed(self, scale_dist: ScalePosition, curve_dist: CurvePosition) -> Self {
+        self.add(None, scale_dist, curve_dist)
+    }
+
+    /// Builds an anchor and adds it to the scale builder
+    /// Distances are relative to previous anchor
+    pub fn add(
+        mut self,
+        id: Option<String>,
+        scale_dist: ScalePosition,
+        curve_dist: CurvePosition,
+    ) -> Self {
+        let last_anchor = self
+            .anchors
+            .last()
+            .expect("The builder should have at least one anchor");
+
+        self.anchors.push(Anchor {
+            id,
+            scale_position: last_anchor.scale_position + scale_dist,
+            curve_position: last_anchor.curve_position + curve_dist,
+        });
+        self
+    }
+
+    /// Requires at least one named anchor
+    /// Will fail if none is present and if the anchors names are duplicated
+    /// This will consume the builder that can not be used after
+    pub fn build(self) -> Result<LrmScale, LrmScaleError> {
+        let mut names = std::collections::HashSet::new();
+        for anchor in self.anchors.iter() {
+            if let Some(name) = &anchor.id {
+                if names.contains(&name) {
```

Comment:
I think we could do something like (might be not exist since i don't know rust syntax):
```rust            
if !names.insert(name) {
    return Err(LrmScaleError::DuplicatedAnchorName(name.to_string()));
}
```

Because a set has unique elements by definition

Reply:
Indeed, much better, rust also returns a bool:
https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.insert